src/sql/workbench/services/accountManagement/browser/accountManagementService.ts

```diff
@@ -221,9 +223,11 @@ export class AccountManagementService implements IAccountManagementService {
 		let self = this;
 
 		// 1) Get the accounts from the store
-		// 2) Update our local cache of accounts
+		// 2) Filter the accounts based on the auth library
+		// 3) Update our local cache of accounts
 		return this.doWithProvider(providerId, provider => {
 			return self._accountStore.getAccountsByProvider(provider.metadata.id)
+				.then(accounts => this._authLibrary ? filterAccounts(accounts, this._authLibrary) : accounts)
```

Comment:
While you're here it would be nice to make this callback async and use await instead

Reply:
Issue https://github.com/microsoft/azuredatastudio/issues/21502 shall track this.